### PR TITLE
gson to jackson

### PR DESCRIPTION
**Pull request by brahimibob**

### CODE REVIEW

File: online-server/src/main/java/eu/itesla_project/online/server/message/ConnectionMessage.java [hunk 1]
```diff
@@ -29,9 +31,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/ConnectionMessage.java]: I'm not sure that it's a good idea to return an empty JSON here. You should probably throw a RuntimeException

File: online-server/src/main/java/eu/itesla_project/online/server/message/RunningMessage.java [hunk 2]
```diff
@@ -29,8 +31,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/RunningMessage.java]: Create a function somewhere to avoid code duplication

File: online-server/src/main/java/eu/itesla_project/online/server/message/SelectedWorkFlowInfoMessage.java [hunk 3]
```diff
@@ -33,8 +35,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/SelectedWorkFlowInfoMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/StableContingenciesMessage.java [hunk 4]
```diff
@@ -33,9 +35,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/StableContingenciesMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithActionsSynthesisMessage.java [hunk 5]
```diff
@@ -31,9 +33,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithActionsSynthesisMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithIndexesSynthesisMessage.java [hunk 6]
```diff
@@ -31,9 +33,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithIndexesSynthesisMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithSecurityRulesResultSynthesisMessage.java [hunk 7]
```diff
@@ -32,9 +34,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/StatesWithSecurityRulesResultSynthesisMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/StatusMessage.java [hunk 8]
```diff
@@ -31,9 +33,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/StatusMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/UnstableContingenciesMessage.java [hunk 9]
```diff
@@ -32,8 +34,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/UnstableContingenciesMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/WcaContingenciesMessage.java [hunk 10]
```diff
@@ -32,8 +34,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/WcaContingenciesMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/WcaRunningMessage.java [hunk 11]
```diff
@@ -31,9 +33,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/WcaRunningMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/WorkFlowIdsMessage.java [hunk 12]
```diff
@@ -32,8 +35,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/WorkFlowIdsMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/WorkStatusMessage.java [hunk 13]
```diff
@@ -29,9 +33,13 @@ protected String getType() {
...
-        Gson gson = new Gson();
-        return gson.toJson(this);
-
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/WorkStatusMessage.java]: Same remark

File: online-server/src/main/java/eu/itesla_project/online/server/message/WorkflowListMessage.java [hunk 14]
```diff
@@ -34,8 +36,13 @@ protected String getType() {
...
     public String toJson() {
-        Gson gson = new Gson();
-        return gson.toJson(this);
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(this);
+        } catch (Exception e) {
+            return "";
+        }
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/WorkflowListMessage.java]: Same remark

File: online-server/src/main/java/resources/application.properties [hunk 15]
```diff
@@ -11,4 +11,4 @@ welcomeTitle=JSF Blank Application
 welcomeHeading=Welcome!
 
 welcomeMessage=This is a JSF blank application. \
-	You can find the application.properties file with this message in the src/resources folder.
+     You can find the application.properties file with this message in the src/resources folder.
```
Comment (mathbagu) [inline online-server/src/main/java/resources/application.properties]: This change is not related to this PR. Don't mix features/fixes

File: online-server/src/main/java/eu/itesla_project/online/server/message/ConnectionMessage.java (hunk 1 above)
Comment (brahimibob) [reply online-server/src/main/java/eu/itesla_project/online/server/message/ConnectionMessage.java]: ok

**Comment by brahimibob** [comment]
done!

File: online-server/src/main/java/eu/itesla_project/online/server/message/Message.java [hunk 16]
```diff
@@ -49,4 +53,14 @@ public String toJson() {
...
         return writer.toString();
     }
+
+    public String writeValueAsString(Object obj) {
+        ObjectMapper json = new ObjectMapper();
+        json.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        try {
+            return json.writeValueAsString(obj);
+        } catch (Exception e) {
+            throw new RuntimeException();
```
Comment (mathbagu) [inline online-server/src/main/java/eu/itesla_project/online/server/message/Message.java]: You have to keep the cause: throw new RuntimeException(e)

File: pom.xml [hunk 17]
```diff
@@ -33,7 +33,7 @@
         <module>eurostag-step-up-transformer</module>
         <module>file-system-rules-db</module>
         <module>fpf-integration</module>
-        <!-- <module>histoDB</module> -->
+         <!--<module>histoDB</module>-->
```
Comment (mathbagu) [inline pom.xml]: Revert this change

File: online-server/src/main/java/resources/application.properties (hunk 15 above)
Comment (mathbagu) [reply online-server/src/main/java/resources/application.properties]: Revert this change
